packages/veritone-react-common/src/components/HeaderBar/index.js

```diff
@@ -0,0 +1,216 @@
+      type: 'modifiedDatetime'
+    }
+  ]
+
+  onOpenSort = (event) => {
+    this.setState({
+      anchorEl: event.currentTarget
+    })
+  }
+
+  clickAway = () => {
+    this.setState({
+      anchorEl: null
+    })
+  }
+
+  render() {
+    const {
+      classes,
+      onUpload,
+      onBack,
+      isStream,
+      viewType,
+      onToggleView,
+      onSort,
+      onSearch,
+      onClear,
+      onCrumbClick,
+      pathList,
+      searchValue
+    } = this.props;
+    const { anchorEl } = this.state;
+    return (
+      <div className={classes.root}>
+        <AppBar position="static" color="default">
+          <Toolbar className={classes.header}>
+            {
+              isStream ? (
+                <BackButton onClick={onBack} />
+              ) : (
+                <UploadButton onClick={onUpload} />
+              )
+            }
+            <div className={classes.spacer} />
+            <Breadcrumbs pathList={pathList} onCrumbClick={onCrumbClick} />
+            <div style={{ flexGrow: 1 }} />
+            <div className={cx(
+              classes.buttonGroup,
+              classes.icon,
+              { [classes.disabled]: isStream }
+              )}
+            >
+              <SearchInput
+                onClear={onClear}
+                onChange={onSearch}
+                value={searchValue}
+              />
+              <div className={classes.spacer} />
+              <div className={cx(classes.divider, classes.icon)} />
+              <div className={classes.spacer} />
+              <Sort onClick={this.onOpenSort} />
+              <ArrowDropDown onClick={this.onOpenSort} />
+              {
+                viewType === 'list' ? (
+                  <ViewList data-type="grid" onClick={onToggleView} />
+                ) : (
+                  <ViewModule data-type="list" onClick={onToggleView} />
+                )
+              }
+            </div>
+          </Toolbar>
+        </AppBar>
+        <Popover
+          open={Boolean(anchorEl)}
+          anchorEl={anchorEl}
+          onClick={this.clickAway}
+          anchorOrigin={{
+            vertical: 'bottom',
+            horizontal: 'left',
+          }}
+          transformOrigin={{
+            vertical: 'top',
+            horizontal: 'left',
+          }}
+        >
+          <MenuList role="menu">
+            {
+              this.sortAbilities.map(({ text, type }) => (
+                <MenuItem key={type} data-type={type} onClick={onSort}>
+                  {text}
+                </MenuItem>
+              ))
+            }
+          </MenuList>
+        </Popover>
+      </div>
+    );
+  }
+}
+
+HeaderBar.propTypes = {
+  classes: shape(Object.keys(styles).reduce((classesShape, key) => ({
+    ...classesShape,
+    [key]: string
+  }), {})),
+  pathList: arrayOf(shape({
+    label: string,
+    icon: string,
+    id: string
+  })),
+  onCrumbClick: func,
+  onSearch: func,
+  onClear: func,
+  onBack: func,
```

Comment:
Since we're not planning to retain a history of the previous views... lets just make the `onBack` be an internal function (not a prop) that will execute `toggleViewType('folder')`